homeassistant/components/yamaha/media_player.py

```diff
@@ -205,6 +205,8 @@ def __init__(self, name, receiver, source_ignore, source_names, zone_names):
         self._play_status = None
         self._name = name
         self._zone = receiver.zone
+        if self.receiver.serial_number is not None:
+            self._attr_unique_id = f"{self.receiver.serial_number}_{self._zone}"
```

Comment:
I looked into the library and integration setup and it's only one out of three possible ways to create the receiver controllers that will set a serial number attribute. It's set when using the library `find` method via ssdp.

I think this is a too low yield and the risk of setting up the integration without a unique id for the entities for users is too great. We can add it like this, but I wouldn't approve changing the default name after this as we don't know if it's a breaking change or not.

I suggest instead starting with building a config flow where we can both manually discover receivers via the library ssdp discovery, and using our built in ssdp discovery. This is a bigger task but will give a greater reward.

Reply:
I believe there are currently two methods:
 - Library SSDP
 - yaml
 
When forcing a static IP with yaml you can also set the name there directly. The unique id is less important in this case because the entity will not change IP and mess things up like it can currently when using SSDP.

I can't really help with the config flow, I don't even know Python. A better default name would be nice to have but not a deal breaker...

Reply:
The third way is the legacy discovery via netdisco.

Ok, then let's merge this, but then we can't change the default name.